core/base/src/main/java/org/eclipse/dataspaceconnector/core/base/ExponentialWaitStrategy.java

```diff
@@ -9,17 +9,17 @@
  *
  *  Contributors:
  *       Microsoft Corporation - initial API and implementation
+ *       Fraunhofer Institute for Software and Systems Engineering - adapt class for negotiation process
  *
  */
 
-package org.eclipse.dataspaceconnector.transfer.core.transfer;
+package org.eclipse.dataspaceconnector.core.base;
 
+import org.eclipse.dataspaceconnector.spi.contract.negotiation.NegotiationWaitStrategy;
 import org.eclipse.dataspaceconnector.spi.transfer.TransferWaitStrategy;
 
-/**
- * Implements an exponential backoff strategy for failed iterations.
- */
-public class ExponentialWaitStrategy implements TransferWaitStrategy {
+public class ExponentialWaitStrategy implements TransferWaitStrategy, NegotiationWaitStrategy {
```

Comment:
Do we still need three interfaces for this concept? Would `WaitStrategy` just be enough? (Eventually it can be moved from the `transfer` package in `spi`

Reply:
Not sure. As the same wait strategy is currently used everywhere, we could probably work with just the `WaitStrategy` interface. But then we'd remove the possibility for extensions to provide a custom wait strategy for either contract negotiation or data transfer, as all new implementations would just be registered under the `WaitStrategy` interface.

Reply:
Could we have a single interface but the ability to define different WaitStrategy configurations via a service?

Reply:
That could work. Should this be part of this PR or a subsequent one?

Reply:
In another one for me would work too.

Reply:
Either way, whichever you think is more manageable.

Reply:
Then let's merge this one first and I'll open an issue for providing a service for the wait strategies.

Reply:
@ndr-brt Are you OK with this, if so, I'll merge it

Reply:
@jimmarino yes, ok!